### PR TITLE
BUG: ensure subclass of override class doesn't segfault.

### PR DESCRIPTION
**Pull request by mhvk**

### CODE REVIEW

File: numpy/core/tests/test_umath.py [hunk 1]
```diff
@@ -1861,8 +1862,8 @@ class A(object):
             def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
                 return self, ufunc, method, inputs, kwargs
 
+        inner1d = ncu_tests.inner1d
```
Comment (mhvk) [inline numpy/core/tests/test_umath.py]: This change and the import on top are just so one can run
```
python3 numpy/core/tests/test_umath.py
```
in an environment where the current version of numpy is installed

File: numpy/core/src/umath/override.c [hunk 2]
```diff
@@ -409,7 +409,8 @@ PyUFunc_CheckOverride(PyUFuncObject *ufunc, char *method,
             /* Check for sub-types to the right of obj. */
             for (j = i + 1; j < noa; j++) {
                 other_obj = with_override[j];
-                if (PyObject_Type(other_obj) != PyObject_Type(obj) &&
+                if (other_obj != NULL &&
+                    PyObject_Type(other_obj) != PyObject_Type(obj) &&
```
Comment (charris) [inline numpy/core/src/umath/override.c]: Should have an extra indent here to clearly separate the boolean expression from the body of the if.

File: numpy/core/tests/test_umath.py (hunk 1 above)
Comment (charris) [inline numpy/core/tests/test_umath.py]: Is there a good reason not to prefer the long, but more informative, `inner1d = np.core.umath_tests.inner1d`?

**Comment by charris** [comment]
Couple of style nits, otherwise LGTM.

**Comment by charris** [comment]
I'll fix the style thingies, as the nep needs whitespace vacuuming also.

**Comment by charris** [comment]
Thanks Marten.

File: numpy/core/tests/test_umath.py (hunk 1 above)
Comment (mhvk) [inline numpy/core/tests/test_umath.py]: For some reason, that did not work with just the `python numpy/core/tests/test_umath.py` -- but possibly my installation was half-screwed or so.

Comment (charris) [inline numpy/core/tests/test_umath.py]: Probably missing `import numpy.core,umath_tests`,

Comment (mhvk) [inline numpy/core/tests/test_umath.py]: I do prefer it, just be sure to import `umath_tests`.

**Comment by charris** [comment]
I've got the style cleanups lined up in a separate PR to avoid cluttering the main PR with irrelevant stuff.

**Pull request merged**